cognite/client/_api/simulators/models.py

```diff
@@ -0,0 +1,437 @@
+        """`Retrieve simulator model <https://developer.cognite.com/api#tag/Simulator-Models/operation/retrieve_simulator_model_simulators_models_byids_post>`_
+        Retrieve a simulator model by ID or external ID
+        Args:
+            id (int | None): The id of the simulator model.
+            external_id (str | None): The external id of the simulator model.
+        Returns:
+            SimulatorModel | None: Requested simulator model
+        Examples:
+            Retrieve simulator model by id:
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> res = client.simulators.models.retrieve(id=1)
+
+            Retrieve simulator model by external id:
+                >>> res = client.simulators.models.retrieve(external_id="model_external_id")
+        """
+        self._warning.warn()
+        identifiers = IdentifierSequence.load(ids=id, external_ids=external_id).as_singleton()
+        return self._retrieve_multiple(
+            list_cls=SimulatorModelList,
+            resource_cls=SimulatorModel,
+            identifiers=identifiers,
+        )
+
+    def retrieve_multiple(
+        self, ids: Sequence[int] | None = None, external_ids: SequenceNotStr[str] | None = None
+    ) -> SimulatorModelList:
+        """`Retrieve simulator models <https://developer.cognite.com/api#tag/Simulator-Models/operation/retrieve_simulator_model_simulators_models_byids_post>`_
+        Retrieve multiple simulator models by IDs or external IDs
+        Args:
+            ids (Sequence[int] | None): IDs.
+            external_ids (SequenceNotStr[str] | None): External Ids.
+        Returns:
+            SimulatorModelList: Requested simulator model
+        Examples:
+            Retrieve simulator model by id:
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> res = client.simulators.models.retrieve(ids=[1,2])
+
+            Retrieve simulator model by external id:
+                >>> res = client.simulators.models.retrieve(external_ids=["model_external_id", "model_external_id2"])
+        """
+        self._warning.warn()
+        identifiers = IdentifierSequence.load(ids=ids, external_ids=external_ids)
+        return self._retrieve_multiple(
+            list_cls=SimulatorModelList,
+            resource_cls=SimulatorModel,
+            identifiers=identifiers,
+        )
+
+    def __iter__(self) -> Iterator[SimulatorModel]:
+        """Iterate over simulator models
+
+        Fetches simulator models as they are iterated over, so you keep a limited number of simulator models in memory.
+
+        Returns:
+            Iterator[SimulatorModel]: yields Simulator model one by one.
+        """
+        return self()
+
+    @overload
+    def __call__(
+        self, chunk_size: int, filter: SimulatorModelsFilter | None = None, limit: int | None = None
+    ) -> Iterator[SimulatorModel]: ...
```

Comment:
This looks wrong. If an int is specified (assuming != 1) `SimulatorModelList` should be returned?

Reply:
Updated overload signature